[BUG] v5.0.0-beta.3 - Thymeleaf template error when trying to add new map style

**Describe the bug**
Opening the map style form in Settings throws a Thymeleaf template error and the form fails to render. The SpringEL expression `style?.dataSource.proxyTiles` in `settings/fragments/map-styles` (line 213) uses safe navigation on `style` but then dereferences `dataSource.proxyTiles` unsafely. When `style.dataSource` is `null` (e.g. a new/empty map style), this throws a `TemplateProcessingException` and the request fails.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to 'Settings' → 'Map Styles'
2. Click to add a new map style (or edit one without a configured data source)
3. The request to `/settings/map-styles/form` is triggered
4. See error — the form does not render

**Expected behavior**
The map style form renders correctly. The `proxyTiles` value should be read safely (e.g. `style?.dataSource?.proxyTiles`) so a missing/null `dataSource` defaults gracefully instead of throwing.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: CachyOS
 - Browser: Brave

**Additional context**
Reitti version: v5.0.0-beta.3

Relevant log output:
```
2026-06-29 13:57:04.933 [ERROR] org.thymeleaf.TemplateEngine - [THYMELEAF][http-nio-8080-exec-44] Exception processing template "settings/fragments/map-styles::[style-form]": Exception evaluating SpringEL expression: "style?.dataSource.proxyTiles" (template: "settings/fragments/map-styles" - line 213, col 28)
2026-06-29 13:57:04.949 [ERROR] org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/].[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "style?.dataSource.proxyTiles" (template: "settings/fragments/map-styles" - line 213, col 28)] with root cause
2026-06-29 13:57:04.964 [ERROR] com.dedicatedcode.reitti.controller.CustomErrorController - Error 200 occurred for request URI: /settings/map-styles/form, message:
```

Note: the safe-navigation operator (`?.`) is applied only to `style`, not to `dataSource`. The chained access `dataSource.proxyTiles` is unguarded, so a `null` `dataSource` is the likely root cause.

--- a/src/main/resources/templates/settings/fragments/map-styles.html
+++ b/src/main/resources/templates/settings/fragments/map-styles.html
@@ -210,7 +210,7 @@
             <div th:if="${isAdmin}">
                 <div class="form-group slide-reveal-container">
                     <input type="checkbox" name="proxyTiles" id="custom-map-style-share-copy"
-                           th:checked="${style?.dataSource.proxyTiles}">
+                           th:checked="${style?.dataSource?.proxyTiles}">
                     <label for="custom-map-style-share-copy" class="slide-reveal">
                         <span class="slide-box"></span>
                         <span class="label-text" th:text="#{map.settings.dialog.map-styles.proxy-tiles}">Proxy tiles through Reitti</span>

